### PR TITLE
[release-1.4] :seedling: Set Kubelet args internally in CAPD

### PR DESCRIPTION
**Pull request by killianmuldoon**
Manual cherry-pick of https://github.com/kubernetes-sigs/cluster-api/pull/8921 and https://github.com/kubernetes-sigs/cluster-api/pull/8881

What this PR does / why we need it:
This PR is a follow up of https://github.com/kubernetes-sigs/cluster-api/pull/8881 automatic generation of another kubeletExtraArg and dropping corresponding code from templates all around.

The PS also includes the cleanup of the kubeletExtraArgs from the cmd/clusterctl/client/cluster/assets folder, which I forgot in the previous PR

### CODE REVIEW

File: test/infrastructure/docker/templates/clusterclass-quick-start.yaml [hunk 1]
```diff
@@ -94,44 +94,6 @@ spec:
         path: "/spec/template/spec/kubeadmConfigSpec/clusterConfiguration/imageRepository"
         valueFrom:
           variable: imageRepository
```
Comment (sbueringer) [inline test/infrastructure/docker/templates/clusterclass-quick-start.yaml:96]: Diffed against main. Should we also drop the replaceImage patches?

**Review by sbueringer — COMMENTED** [review]
Diffed the PR against main. Almost everything as expected. One question about the ClusterClass

File: test/infrastructure/docker/templates/clusterclass-quick-start.yaml (hunk 1 above)
Comment (fabriziopandini) [reply test/infrastructure/docker/templates/clusterclass-quick-start.yaml:96]: I would say that the patch changing the image repository in clusterConfiguration is orthogonal on this issue; also, if we want to change something here, we have to change it in main first

Comment (sbueringer) [reply test/infrastructure/docker/templates/clusterclass-quick-start.yaml:96]: We changed it on main already. I found this by diffing against main

Comment (sbueringer) [reply test/infrastructure/docker/templates/clusterclass-quick-start.yaml:96]: For reference: https://github.com/kubernetes-sigs/cluster-api/pull/8921/files#diff-813c2a40b85d2d2a6737e5f20041314716efaf113af26c458f50912f764fa4a4

Comment (killianmuldoon) [reply test/infrastructure/docker/templates/clusterclass-quick-start.yaml:96]: I think I've removed the patch you were talking about now - let me know if I missed something.

Comment (sbueringer) [reply test/infrastructure/docker/templates/clusterclass-quick-start.yaml:96]: Yup, looks perfect!

**Pull request merged**